### PR TITLE
Enable feature set backfill with jobId and dataStatus

### PR DESCRIPTION
**Pull request by fredms**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py [hunk 1]
```diff
@@ -26,15 +26,15 @@ class FeatureSetBackfillMetadata(RestTranslatableMixin):
     def __init__(
         self,
         *,
-        job_id: Optional[str] = None,
+        job_ids: Optional[list] = None,
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py:29]: For my understanding: why the schema takes a single job_id but here we take a list of job_ids?
Also, type hint for collection should include item type, e.g Optional[List[str]]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py [hunk 2]
```diff
@@ -178,13 +178,15 @@ def begin_backfill(
         *,
         name: str,
         version: str,
-        feature_window_start_time: datetime,
-        feature_window_end_time: datetime,
+        feature_window_start_time: Optional[datetime] = None,
```
Comment (bastrik) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:181]: If start time/end time is not provided, what will be the backfill time window?

Comment (fredms) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:181]: it should be based on job id or data status. but @shail2208 can add more insights.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py (hunk 1 above)
Comment (fredms) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py:29]: backend service supports list of ids in the new swagger.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py (hunk 2 above)
Comment (shail2208) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_set_operations.py:181]: Logic is like this
1.  --by-data-status None, Incomplete --start-time --end-time
2. --by-data-status None, Incomplete  (start_time and end time is inferred from the materialization status, start time is the minimum of all the materialized window and end time is the max of all materialized window till now)
3. --by-job-id (start time and end time should not be given) (start and end time are inferred from the already executed job)

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py (hunk 1 above)
Comment (shail2208) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py:29]: Backfill request

take single job id => Which means retry backfilling the interval for what an already executed materialization job ran for.

Backfill response
return multiple job ids (list of all jobs that were submitted as part of this backfill request.

Comment (bastrik) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py:29]: I see, so `feature_set_backfill_metadata` is the response object, and `feature_set_backfill_schema`/`feature_set_backfill_request` are user input?

Comment (fredms) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_backfill_metadata.py:29]: you're right.

**Review by bastrik — APPROVED** [review]

**Review by qjxu — APPROVED** [review]

**Pull request merged**